uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-actions.component.js

```diff
@@ -0,0 +1,224 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components',
+  'common/services/hr-settings'
+], function (_, moment, components) {
+  components.component('leaveRequestActions', {
+    bindings: {
+      leaveRequest: '<',
+      leaveRequestStatuses: '<',
+      absenceTypes: '<',
+      /**
+       * Role is not a permission level in this case.
+       * For example Manager can act as Staff
+       * and Admin can act as either Manager or Staff.
+       */
+      role: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-request-actions.html';
+    }],
+    controllerAs: 'actions',
+    controller: ['$log', '$q', '$rootScope', 'OptionGroup', 'dialog', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, OptionGroup, dialog, sharedSettings) {
+    $log.debug('Component: leave-request-action-dropdown');
+
+    var vm = this;
+    var actions = {
+      edit: {
+        label: 'Edit',
+        statuses: [sharedSettings.statusNames.awaitingApproval]
```

Comment:
if you store the reference in a variable, you can shorten the lines and make the code easier to read;
```js
var statusNames = shared.Settings.statusNames;

// ...

var actions = {
  edit: {
    statuses: [statusNames.awaitingApproval]
  }
}